src/commands/ban.ts

```diff
@@ -12,11 +12,20 @@ async function banHandler(interaction: ChatInputCommandInteraction): Promise<voi
 
 	const guild = await interaction.guild!.fetch();
 	const executor = interaction.user;
-	const users = interaction.options.getString('users', true);
+	const subcommand = interaction.options.getSubcommand();
 	const reason = interaction.options.getString('reason', true);
 	const deleteMessages = interaction.options.getNumber('delete_messages');
 
-	const userIds = [...new Set(Array.from(users!.matchAll(new RegExp(MessageMentions.UsersPattern, 'g')), match => match[1]))];
+	let userIds;
+	if (subcommand === 'user') {
+		const user = interaction.options.getUser('user', true);
+		userIds = [user.id];
+	} else if (subcommand === 'multiuser') {
+		const users = interaction.options.getString('users', true);
+		userIds = [...new Set(Array.from(users.matchAll(/\d{17,18}/g), match => match[0]))];
+	} else {
+		throw new Error(`Unknown ban subcommand: ${subcommand}`);
+	}
```

Comment:
```suggestion
	let userIDs;
	if (subcommand === 'user') {
		const user = interaction.options.getUser('user', true);
		userIDs = [user.ID];
	} else if (subcommand === 'multiuser') {
		const users = interaction.options.getString('users', true);
		userIDs = [...new Set(Array.from(users.matchAll(/\d{17,18}/g), match => match[0]))];
	} else {
		throw new Error(`Unknown ban subcommand: ${subcommand}`);
	}
```

Reply:
Partially implemented because discord.js uses `id` (so `user.ID` is not valid).